lib/metrics/amplitude.js

```diff
@@ -0,0 +1,200 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const APP_VERSION = /^[0-9]+\.([0-9]+)\./.exec(require('../../package.json').version)[1]
+
+const GROUPS = {
+  activity: 'fxa_activity',
+  email: 'fxa_email',
+  login: 'fxa_login',
+  registration: 'fxa_reg',
+  sms: 'fxa_sms'
+}
+
+const EVENTS = {
+  'account.confirmed': {
+    group: GROUPS.login,
+    event: 'email_confirmed'
+  },
+  'account.created': {
+    group: GROUPS.registration,
+    event: 'created'
+  },
+  'account.login': {
+    group: GROUPS.login,
+    event: 'success'
+  },
+  'account.login.blocked': {
+    group: GROUPS.login,
+    event: 'blocked'
+  },
+  'account.login.confirmedUnblockCode': {
+    group: GROUPS.login,
+    event: 'unblock_success'
+  },
+  'account.reset': {
+    group: GROUPS.login,
+    event: 'forgot_complete'
+  },
+  'account.signed': {
+    group: GROUPS.activity,
+    event: 'cert_signed'
+  },
+  'account.verified': {
+    group: GROUPS.registration,
+    event: 'email_confirmed'
+  },
+  'flow.complete': {
+    isDynamicGroup: true,
+    group: (request, data, metricsContext) => GROUPS[metricsContext.flowType],
+    event: 'complete'
+  },
+  'password.forgot.resend_code.completed': {
+    group: GROUPS.login,
+    event: 'forgot_sent'
+  },
+  'password.forgot.send_code.completed': {
+    group: GROUPS.login,
+    event: 'forgot_sent'
+  },
+  'sms.installFirefox.sent': {
+    group: GROUPS.sms,
+    event: 'sent'
+  }
+}
+
+const EMAIL_EVENTS = /^email\.(\w+)\.(bounced|sent)$/
+
+const EMAIL_TYPES = {
+  newDeviceLoginEmail: 'login',
+  passwordChangedEmail: 'change_password',
+  passwordResetEmail: 'reset_password',
+  passwordResetRequiredEmail: 'reset_password',
+  postRemoveSecondaryEmail: 'secondary_email',
+  postVerifyEmail: 'registration',
+  postVerifySecondaryEmail: 'secondary_email',
+  recoveryEmail: 'reset_password',
+  unblockCode: 'unblock',
+  verificationReminderFirstEmail: 'registration',
+  verificationReminderSecondEmail: 'registration',
+  verificationReminderEmail: 'registration',
+  verifyEmail: 'registration',
+  verifyLoginEmail: 'login',
+  verifySyncEmail: 'registration',
+  verifySecondaryEmail: 'secondary_email'
+}
+
+const NOP = () => {}
+
+const EVENT_PROPERTIES = {
+  [GROUPS.activity]: NOP,
+  [GROUPS.email]: mapEmailType,
+  [GROUPS.login]: NOP,
+  [GROUPS.registration]: NOP,
+  [GROUPS.sms]: NOP
+}
+
+const USER_PROPERTIES = {
+  [GROUPS.activity]: mapUid,
+  [GROUPS.email]: mapUid,
+  [GROUPS.login]: mapUid,
+  [GROUPS.registration]: mapUid,
+  [GROUPS.sms]: NOP
+}
+
+module.exports = log => {
+  if (log) {
```

Comment:
I don't expect it to, just did it so things fail faster if it ever gets forgotten.